pkg/apis/core/validation/validation.go

```diff
@@ -3352,14 +3352,116 @@ func validateHandler(handler commonHandler, gracePeriod *int64, fldPath *field.P
 	return allErrors
 }
 
-func validateLifecycle(lifecycle *core.Lifecycle, gracePeriod *int64, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+var supportedStopSignalsLinux = sets.New(
+	core.Sigabrt,
+	core.Sigalrm,
+	core.Sigbus,
+	core.Sigchld,
+	core.Sigcld,
+	core.Sigcont,
+	core.Sigfpe,
+	core.Sighup,
+	core.Sigill,
+	core.Sigint,
+	core.Sigio,
+	core.Sigiot,
+	core.Sigkill,
+	core.Sigpipe,
+	core.Sigpoll,
+	core.Sigprof,
+	core.Sigpwr,
+	core.Sigquit,
+	core.Sigsegv,
+	core.Sigstkflt,
+	core.Sigstop,
+	core.Sigsys,
+	core.Sigterm,
+	core.Sigtrap,
+	core.Sigtstp,
+	core.Sigttin,
+	core.Sigttou,
+	core.Sigurg,
+	core.Sigusr1,
+	core.Sigusr2,
+	core.Sigvtalrm,
+	core.Sigwinch,
+	core.Sigxcpu,
+	core.Sigxfsz)
+
+var supportedStopSignalsWindows = sets.New(core.Sigkill, core.Sigterm)
+
+func validateStopSignal(stopSignal *core.Signal, fldPath *field.Path, os *core.PodOS) field.ErrorList {
+	allErrors := field.ErrorList{}
+
+	if os == nil {
```

Comment:
I think we can drop this and assume linux as default.
cc: @thockin

Reply:
I really want to start requiring explicit OS indication for use of OS-specific fields

Reply:
So we should validate this as "if you didn't say Linux, you don't get to use Linux-isms?"  I think I am OK with that.

Reply:
to use new OS-specific fields, yeah, I'd lean that way

Reply:
I've added the note in https://github.com/kubernetes/kubernetes/pull/130556/commits/caeeb362b68e3db945bc68c68d45829636ab7746. PTAL!